raiden/messages/metadata.py

```diff
@@ -80,6 +89,7 @@ class Metadata:
     # the initiator of a transfer.
     original_data: Optional[Any] = None
     encrypted_secret: Optional[EncryptedSecret] = None
+    _hash: Optional[MetadataHash] = None
```

Comment:
This is not good!

The problem is that `marshmallow_dataclass` will  include this in the schema as soon as it is defined here.
If a sender adds a `_hash` field to the metadata, he has control over your validation process and can chose part of your assumed hash for the locked-transfer and therefore (possibly) forge a message. 

Why don't you call the `hash_metadata_v2_0_0` here and use the `_hash` as a flag that switches the behaviour? 
You probably additionally / alternatively should check for existence of the field in the incoming message.

Reply:
very good catch! you are totally right. There should be at least a check if `_hash` is in the message and should be removed on the `post_load`. But the flag sounds like a better idea after all.